lib/shelley/src/Cardano/Wallet/Shelley/Network.hs

```diff
@@ -456,10 +456,10 @@ withNetworkLayer tr np addrInfo versionData action = do
             bracketTracer (contramap (MsgWatcherUpdate header) tr) $
                 cb header
 
-    _timeInterpreterQuery :: HasCallStack => TVar IO (Maybe (CardanoInterpreter sc)) -> TimeInterpreter IO
+    _timeInterpreterQuery :: HasCallStack => MVar (CardanoInterpreter sc) -> TimeInterpreter IO
     _timeInterpreterQuery var query = do
-        cached <- atomically (readTVar var)
-        let interpret = maybe (interpreterFromGenesis gp) (mkTimeInterpreter getGenesisBlockDate) cached
+        cached <- takeMVar var
+        let interpret = (mkTimeInterpreter getGenesisBlockDate) cached
```

Comment:
I am quite failing to grasp the consequences of this.

Reply:
Goal:
```
On launch, the TimeInterpreter may not have been fetched from the node.

Instead returning the singleEraInterpreter for the first era, it seems
safer to block until fetched.

I imagine there could otherwise be race conditions where we would sometimes
return completely wrong time data when in Shelley, just after starting the
node.
```
But:
- I should use `readMVar` instead of `takeMVar` 😅